metals/src/main/scala/scala/meta/internal/metals/codeactions/ActionableDiagnostic.scala

```diff
@@ -19,35 +20,54 @@ class ActionableDiagnostic() extends CodeAction {
 
     def createActionableDiagnostic(
         diagnostic: l.Diagnostic,
-        textEdit: l.TextEdit,
+        changes: Map[AbsolutePath, List[l.TextEdit]],
     ): l.CodeAction = {
       val diagMessage = diagnostic.getMessage
-      val uri = params.getTextDocument().getUri()
 
       CodeActionBuilder.build(
         title =
           s"Apply suggestion: ${diagMessage.linesIterator.headOption.getOrElse(diagMessage)}",
         kind = l.CodeActionKind.QuickFix,
-        changes = List(uri.toAbsolutePath -> Seq(textEdit)),
+        changes = changes.toSeq,
         diagnostics = List(diagnostic),
       )
     }
 
-    val codeActions = params
+    val codeActions: List[l.CodeAction] = params
       .getContext()
       .getDiagnostics()
       .asScala
       .groupBy {
-        case ScalacDiagnostic.ScalaAction(textEdit) =>
-          Some(textEdit)
-        case _ => None
+        case ScalacDiagnostic.DiagnosticData(data) =>
+          Some(data)
+        case _ =>
+          None
       }
       .collect {
-        case (Some(textEdit), diags)
+        case (Some(Left(textEdit)), diags)
             if params.getRange().overlapsWith(diags.head.getRange()) =>
-          createActionableDiagnostic(diags.head, textEdit)
+          List(
+            createActionableDiagnostic(
+              diags.head,
+              Map(
+                params.getTextDocument().getUri().toAbsolutePath -> List(
+                  textEdit
+                )
+              ),
+            )
+          )
+        case (Some(Right(edits)), diags)
+            if params.getRange().overlapsWith(diags.head.getRange()) =>
+          for {
+            edit <- edits.edits.asScala.toList
```

Comment:
```suggestion
            edit <- diagnosticData.edits.asScala.toList
```
maybe? So that we don't get the weird `edit.edit` ?